src/com/oltpbenchmark/api/Worker.java

```diff
@@ -402,11 +410,8 @@ public final void run() {
                 }
             }
             // In case of transaction failures, the end times will not be populated.
-            if (executionState.getEndOperation() == 0) {
-              executionState.setEndOperation(System.nanoTime());
-            }
-            if (executionState.getEndConnection() == 0) {
-              executionState.setEndConnection(System.nanoTime());
+            if (executionState.getEndOperation() == 0 || executionState.getEndConnection() == 0) {
```

Comment:
do we ever actually get here? and if so, should we be throwing an exception instead of continuing?

Reply:
yup.
2 cases,
1.  when we don't get a connection from the Hikari connection pool.
2.  When the transaction fails for any reason.

In those cases, we will come here. 
Now we can continue to execute or fail.. I chose to continue. We can fail here as well.